apps/api/src/database/migrations/1730724383210-ArchiveCompletedNotifications.ts

```diff
@@ -0,0 +1,221 @@
+          },
+          {
+            name: 'updated_on',
+            type: 'timestamp',
+            default: 'CURRENT_TIMESTAMP',
+            onUpdate: 'CURRENT_TIMESTAMP',
+          },
+          {
+            name: 'created_by',
+            type: 'varchar',
+          },
+          {
+            name: 'updated_by',
+            type: 'varchar',
+          },
+          {
+            name: 'status',
+            type: 'tinyint',
+            default: 1,
+          },
+          {
+            name: 'application_id',
+            type: 'int',
+            isNullable: true,
+          },
+          {
+            name: 'provider_id',
+            type: 'int',
+            isNullable: true,
+          },
+          {
+            name: 'retry_count',
+            type: 'int',
+            default: 0,
+          },
+        ],
+      }),
+    );
+
+    // Create foreign keys for table notify_archived_notifications
+    await queryRunner.createForeignKey(
+      'notify_archived_notifications',
+      new TableForeignKey({
+        columnNames: ['channel_type'],
+        referencedColumnNames: ['master_id'],
+        referencedTableName: 'notify_master_providers',
+        onDelete: 'CASCADE',
+      }),
+    );
+
+    await queryRunner.createForeignKey(
+      'notify_archived_notifications',
+      new TableForeignKey({
+        columnNames: ['provider_id'],
+        referencedColumnNames: ['provider_id'],
+        referencedTableName: 'notify_providers',
+        onDelete: 'CASCADE',
+      }),
+    );
+
+    // Add indexes to table notify_archived_notifications
+    await queryRunner.createIndex(
+      'notify_archived_notifications',
+      new TableIndex({
+        name: 'IDX_ARCHIVED_NOTIFICATION_ID',
+        columnNames: ['notification_id'],
+      }),
+    );
+
+    await queryRunner.createIndex(
+      'notify_archived_notifications',
+      new TableIndex({
+        name: 'IDX_ARCHIVED_DELIVERY_STATUS',
+        columnNames: ['delivery_status'],
+      }),
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    const conflicts = await queryRunner.query(`
+      SELECT an.notification_id
+      FROM notify_archived_notifications an
+      JOIN notify_notifications nn ON an.notification_id = nn.id
+   `);
+
+    if (conflicts.length > 0) {
+      throw new Error(`Found ${conflicts.length} ID conflicts in notify_notifications table`);
+    }
+
+    // Copy the entries in notify_archived_notifications back to notify_notifications
+    await queryRunner.query(`
+      INSERT INTO notify_notifications (
+        id, channel_type, data, delivery_status, result, created_on, updated_on,
+        created_by, updated_by, status, application_id, provider_id, retry_count
+      ) SELECT
+        notification_id, channel_type, data, delivery_status, result, created_on, updated_on,
+        created_by, updated_by, status, application_id, provider_id, retry_count
+      FROM notify_archived_notifications
+    `);
+
```

Comment:
@kshitij-k-osmosys  Check what this is.

Reply:
This should ideally not happen since I am also 1) storing the original and 2) inserting back the same stored ID value

While running migrations on local did not see tthis conflict happen

Also have a check for conflicts at line 150 here https://github.com/OsmosysSoftware/osmo-x/pull/360/files#diff-4d0b3200f1ddd8d98a2a7db82227820f1b54e82a96e70132abba9b5360469518R150